pkg/operator/rules.go

```diff
@@ -195,10 +195,24 @@ func ValidateRule(promRuleSpec monitoringv1.PrometheusRuleSpec) []error {
 		return []error{fmt.Errorf("the length of rendered Prometheus Rule is %d bytes which is above the maximum limit of %d bytes", promRuleSize, MaxConfigMapDataSize)}
 	}
 
-	_, errs := rulefmt.Parse(content, false)
+	_, errs := rulefmt.Parse(content, false, validationScheme)
 	return errs
 }
 
+func (prs *PrometheusRuleSelector) getValidationScheme() model.ValidationScheme {
+	// For Prometheus/PrometheusAgent 3.0.0+, use UTF-8 validation
+	// For ThanosRuler, always use legacy validation (ThanosRuler doesn't support UTF-8 yet)
+	if prs.ruleFormat == ThanosFormat {
+		return model.LegacyValidation
+	}
+
+	if prs.version.GTE(semver.MustParse("3.0.0")) {
```

Comment:
should we factorize with `getValidationSchemeForVersion()` from pkg/prometheus/resource_selector.go?